src/main/main.ts

```diff
@@ -219,13 +219,16 @@ async function initializeApp() {
         .then((name) => console.log(`Added Extension:  ${name}`))
         .catch((err) => console.log("An error occurred: ", err));
 
-    if (app.commandLine.hasSwitch("v2") || getConfig("UseV2Interface"))
-      win = await createV2Window();
+    if (isV2) win = await createV2Window();
     else win = await createWindow();
     createTray(path.join(app.getAppPath(), logoImage));
 
     const u = await checkForUpdates(standalone);
-    if (u) update(u, updateOptions, win);
+    if (u && !isV2) update(u, updateOptions, win);
+    else if (u && isV2)
+      ipcMain.handle("start update", async () => {
+        await update(u, updateOptions, win!);
```

Comment:
What is the difference between line 230 and line 227's `update()` method? why do not have `!` on the 227?

Reply:
It runs the update process immediately on the V1 launcher as this matches the current behaviour. On the V2 it was difficult to start the update process after the event listeners are initialized correctly, so instead of starting it immediately I changed to let V2 frontend trigger it [here](https://github.com/BasixKOR/9c-launcher/blob/d1978aadab42a365748bf5533f1209e9cc267c7c/src/v2/machines/updateMachine.ts#L48)

Reply:
I expect to altogether remove this branching after the V1 launcher sunsetting, but for now, for the backward compatibility, this would do the trick.